### ISSUE TITLE
Program crashes on copy and paste

### ISSUE THREAD

**Issue by AuroraMartell** [olivierkes/manuskript#441]
Sometimes, manuskript unexpectedly quits after using Edit - Paste (or Ctrl + V). I'm fairly sure it's a bug.

**Comment by AuroraMartell** [comment]
Here's what I get in the console log after the program quits on trying to paste some text in the editor.

`Debug: Web rendering engine used: QWebView
Running manuskript version 0.8.0.
Loaded translation: /usr/share/manuskript/i18n/manuskript_pt_BR.qm.
Loading: /home/laura/Documentos/Wanderer.msk
Detected file format version: 1. Zip: True.
Project /home/laura/Documentos/Wanderer.msk loaded.
Fatal Python error: Segmentation fault
`

**Comment by gedakc** [comment]
Thanks for posting the console log.  It contains much less information than I had expected.

It is possible that there is an issue with the language translation file.  To test if this is the case would you be able change the language in **Edit -> Settings -> General -> Language** to **English**.  Stop and then start manuskript (this should start manuskript with English).  Then try to load the project again to see if the segmentation fault occurs.

Please also post the console log when loading the project with the **English** language selected.

**Comment by AuroraMartell** [comment]
I changed the language to English and this is what I get:

```
Debug: Web rendering engine used: QWebView
Running manuskript version 0.8.0.
Loaded translation: /usr/share/manuskript/i18n/manuskript_pt_BR.qm.
Loading: /home/laura/Documentos/Roger & Brianne.msk
Detected file format version: 1. Zip: True.
Project /home/laura/Documentos/Roger & Brianne.msk loaded.
Project Roger & Brianne.msk saved.
Project Roger & Brianne.msk saved.
Fatal Python error: Segmentation fault

```

**Comment by gedakc** [comment]
It looks like Manuskript still "_Loaded translation: /usr/share/manuskript/i18n/manuskript_pt_BR.qm_".  If I recall this is fixed in the **develop** branch with issue #411, but not in the **0.8.0** release.

Can you rename `/usr/share/manuskript/i18n/manuskript_pt_BR.qm` to something different so that it does not get loaded and then try again?

Also can you provide an example what you are copying and pasting that causes the segmentation fault?

**Comment by AuroraMartell** [comment]
I'm copying text (I'm working on a story during school break), but I think this is caused when the user clicks paste (using the context menu) without clicking inside the document.

**Comment by gedakc** [comment]
So far I'm unable to reproduce the issue.

I've tried copying some text within manuskript in the **General** area, changing to another area (e.g., **Plots**), selecting the **Editor** area and then using the menu **Edit -> Paste** but no crash.

If I select a scene, right click and select **paste** then the actual text is pasted with no crash.

Please provide the exact steps you use to make the program crash so that I can try to replicate it.

**Comment by AuroraMartell** [comment]
The issue happens only in the Editor, not anywhere else.

**Comment by gedakc** [comment]
How are you able to invoke the context menu (right-click) in a scene, but not have the cursor set it's position in the scene and ultimately paste the text into the scene?

**Comment by AuroraMartell** [comment]
The way it happens to me is like this: I'd create a couple of documents and copy their contents, delete them, create new documents and use the group button to try to paste them, then the crash happens.

**Comment by gedakc** [comment]
What do you mean by "documents"?  Do you mean "Folders/Chapters" or "Text/Scenes"?

What "group" button are you using?

**Comment by AuroraMartell** [comment]
Texts. I'm using the square button just next to the arrow button in the footer bar.

**Comment by gedakc** [comment]
So far I am still unable to reproduce the crash.  The steps I used were:

1. Start manuskript and select *Empty Fiction* and click **Create**.
2. Provide the project name as "test" and click **Save**.
3. Select **Editor**.
4. Click the add text button (plus square icon) three times to create three new texts.
5. Select each text and type some words into each one.
6. Use *Shift+Click* to select all three texts.
7. *Right+Click* and select **Copy**.
8. Click the remove texts button (minus sign) to delete the three texts and click **Yes** to confirm deletion.
9. *Right+Click* where texts were removed and select **Paste**.

Result:  The three texts are pasted into the project with no crash.

What are you doing differently?

**Comment by AuroraMartell** [comment]
I dunno, really. It just happens on me when I'm working on my stuff. I really wish I could capture a gif of what happens but I can't

**Comment by AuroraMartell** [comment]
I'm running Linux at the moment.

**Comment by gedakc** [comment]
For GNU/Linux you might investigate [Peek](https://github.com/phw/peek) for recording Screens to GIF.

**Comment by AuroraMartell** [comment]
I'm running Ubuntu 18.10.

**Comment by gedakc** [comment]
I tried running the above mentioned steps to create, copy, delete, and paste texts in Ubuntu 18.10 and it did not crash in my VM.

Following is a screen to GIF recording capture using [Peek](https://github.com/phw/peek):

![manuskript-create-copy-delete-paste-texts-ubuntu18 10](https://user-images.githubusercontent.com/10405019/50178535-0efec780-02c2-11e9-9b83-43ceb472148b.gif)

Can you see anything different here than what you are doing?

If so, what?

**Comment by AuroraMartell** [comment]
I'm copying the text _inside_ the documents, not the documents themselves. Copying the documents themselves don't cause any crashing

**Comment by gedakc** [comment]
On Ubuntu 18.10 I tried copying and pasting the text inside documents.  Still no crash.

Here is the screen to GIF recording.

![manuskript-copy-paste-text-ubuntu18 10](https://user-images.githubusercontent.com/10405019/50179937-8550f900-02c5-11e9-8470-163ed47534da.gif)

**Comment by AuroraMartell** [comment]
Try using the simple mode, not the fiction one. Maybe you'll be able to replicate the bug.

**Comment by gedakc** [comment]
I tried coping and pasting text in **Simple** view mode and there was no crash.

Following is the screen to GIF recording of copying a range of characters and pasting into another text/scene.

![manuskript-simple-mode-copy-paste-text-ubuntu18 10](https://user-images.githubusercontent.com/10405019/50185466-124f7e80-02d5-11e9-9048-cf91b0351265.gif)

I am running out of ideas to try.  I think our next step is to have you record the exact steps you are using that cause the crash.

**Comment by AuroraMartell** [comment]
I'll try using the develop branch .deb file and report back if the odd crash still happens, since this whole screen recording software is pretty confusing.

**Comment by gedakc** [comment]
Okay.  If/when you discover a series of steps that reliably reproduces the crash then please list the exact steps in this report.

**Comment by AuroraMartell** [comment]
I will.

**Comment by AuroraMartell** [comment]
As I described earlier, select an empty document and click paste. The cursor will freeze and then Manuskript crashes.

**Comment by gedakc** [comment]
I tried the steps you suggested and still no crash.  Following is the screen to GIF recording made on Ubuntu 18.10.

![manuskript-copy-create-paste-text-ubuntu18 10](https://user-images.githubusercontent.com/10405019/50238503-c3f5ba80-037c-11e9-9a0c-8bfc4dc4d6eb.gif)

If you can get the **develop** branch running on your computer, you might try running manuskript with Python *debug* on to try to isolate where the segmentation fault occurs.

To do this, change into the source code directory for Manuskipt and run the `make debug` command.
When the GNU Debugger starts with prompt `(gdb)` type `run` and then try to reproduce the crash.  The debugging output should hopefully point to where the segmentation fault occurs.

**Comment by TheJackiMonster** [comment]
This should be fixed with the PR #706 which solves the bugs which were caused by parallel data access.
